ismatec/driver.py

```diff
@@ -55,39 +57,36 @@ async def __aexit__(self, *args):
         """Provide exit to the context manager."""
         self.hw._stop_event.set()
 
-    ####################################################################
-    # Properties or setters/getters                                    #
-    # one per channel for the ones that have the channel kwarg.        #
-    ####################################################################
-
-    async def get_pump_version(self) -> str:
-        """Return the pump model, firmware version, and pump head type code."""
-        return self.hw.query('1#').strip()
-
-    async def get_serial_protocol_version(self) -> int:
-        """Return serial protocol version."""
-        return int(self.hw.query('1x!'))
-
-    async def set_flowrate(self, channel: int, flowrate):
-        """Set the flowrate of the specified channel."""
+    async def get_version(self) -> str:
+        """Get device version information."""
+        pump_version = self.hw.query('1#').strip().split(' ')
+        protocol_version = int(self.hw.query('1x!'))
+        return {
+            'description': pump_version[0],
+            'software': pump_version[1],
+            'pump head': pump_version[2],
+            'protocol': protocol_version,
+        }
+
+    async def set_flow_rate(self, channel: int, flowrate):
+        """Set the flow rate of the specified channel."""
         assert channel in self.channels
-        flow = self._volume2(flowrate)
+        flow = self._volume2(flowrate)  # TODO Clearer explanation of datatypes
         self.hw.command(f'{channel}f{flow}')
 
-    async def get_flowrate(self, channel: int) -> float:
-        """Return the current flowrate of the specified channel."""
+    async def get_flow_rate(self, channel: int) -> float:
+        """Get the flow rate of the specified channel."""
         assert channel in self.channels
         reply = self.hw.query(f'{channel}f')
         return float(reply) / 1000 if reply else 0
 
-    async def get_running(self, channel: int) -> bool:
-        """Return True if the specified channel is running."""
+    async def is_running(self, channel: int) -> bool:
+        """Return if the specified channel is running."""
         assert channel in self.channels
-        # self.hw.running[channel] = self.hw.query(f'{channel}E') == '+'
```

Comment:
I know you explained the fact you can't easily tell if the device is running. That's sad and I don't have a good answer. Caching works until it doesn't, but is better than not working at all.

Can you query flow rate or rpm or something to get if running? Will flow be 0 if not running and != 0 if running?

Reply:
Querying RPM just gives the RPM setpoint, sadly